### PR TITLE
Append user token to the audience action iframe url

### PR DESCRIPTION
**Pull request by longmelon**

### CODE REVIEW

File: src/runtime/audience-action-flow.js [hunk 1]
```diff
@@ -91,17 +91,25 @@ export class AudienceActionFlow {
...
-        'supportsEventManager': true,
-        'productType': this.productType_,
-      }),
-      /* shouldFadeBody */ true
-    );
+    /** @private @const {!Promise<?ActivityIframeView>} */
+    this.activityIframeViewPromise_ = deps
+      .storage()
+      .get(Constants.USER_TOKEN, true)
+      .then((token) => {
```
Comment (jpettitt) [inline src/runtime/audience-action-flow.js:98]: storage().get() can return NULL which will result in sud=NULL in your URL - is the your intent and can the backend cope with that?

[we can't guarantee that some extension hasn't trashed storage or if we are in an iframe on safari storage doesn't persist across sessions]

Comment (longmelon) [reply src/runtime/audience-action-flow.js:98]: In line 168, if storage().get() returns null, we will set empty string to the sut param.

**Review by jpettitt — APPROVED** [review]

File: src/runtime/audience-action-flow.js [hunk 2]
```diff
@@ -91,17 +91,25 @@ export class AudienceActionFlow {
...
-      feArgs({
-        'supportsEventManager': true,
-        'productType': this.productType_,
-      }),
-      /* shouldFadeBody */ true
-    );
+    /** @private @const {!Promise<?ActivityIframeView>} */
+    this.activityIframeViewPromise_ = deps
+      .storage()
+      .get(Constants.USER_TOKEN, true)
```
Comment (hartmannr76) [inline src/runtime/audience-action-flow.js:97]: I think you can actually just pass it into the ctor. we have the entitlements response when we create this flow

File: src/runtime/audience-action-flow.js [hunk 3]
```diff
@@ -110,45 +118,50 @@ export class AudienceActionFlow {
...
-    this.activityIframeView_.on(CompleteAudienceActionResponse, (response) =>
-      this.handleCompleteAudienceActionResponse_(response)
-    );
-
-    this.activityIframeView_.on(
-      AlreadySubscribedResponse,
-      this.handleLinkRequest_.bind(this)
-    );
-
-    const {fallback} = this.params_;
```
Comment (hartmannr76) [inline src/runtime/audience-action-flow.js:126]: 👍

**Review by adolgachev — APPROVED** [review]

File: src/runtime/audience-action-flow.js (hunk 2 above)
Comment (longmelon) [reply src/runtime/audience-action-flow.js:97]: Looks like the entitlements object here doesn't contain the user token.

**Pull request merged**